### DISCUSSION TITLE
How to differentiate PV and from Battery production?

### DISCUSSION THREAD

**Discussion by alarmatwork** [StephanJoubert/home_assistant_solarman#688] [Q&A] [answered]
After installing batteries to Deye 15kW HV inverter all my calculations are a bit messed up as discharging from Battery is reported as productiin. Is there a way to get pure solar generated part of production?

I was using a template like this to calculate:

template:
    - sensor:
      - name: "Aex Current Production Sum"
        unique_id: current_production_sum
        state: >-
          {{ 
            [ states('sensor.deye_inverter_l1_power'), 
              states('sensor.deye_inverter_l2_power'),
              states('sensor.deye_inverter_l3_power')
            ] | map('float') | sum
          }}
        availability: >
          {{ not 'unavailable' in 
            [ states('sensor.deye_inverter_l1_power'), 
              states('sensor.deye_inverter_l2_power'),
              states('sensor.deye_inverter_l2_power') ] 
          }}
        unit_of_measurement: W
        device_class: energy
        state_class: measurement

**Comment by davidrapan** [comment]
What are you trying to achieve? There are PV Power sensors.

**Comment by alarmatwork** [comment]
Hi, 

I would like to display Solar Panels power current production on my dashboard. 
It was workig fine when I was summing all deye_inverter_l1_power values, but now as I have batteries connected to my inverter, this value is combination of solar produced energy and energy used from batteries.

I kind of assumed that  PV1 Power etc. sensors are energy I get from the roof, not after conversion, maybe I'm wrong. I will try when there is some solar prodcutio next time.

**Reply by davidrapan** [reply, marked as answer]
That's correct, it's power before AC/DC. Okay so if `deye_inverter_lx_power` is output of the inverter (I'm not sure cause I'm using different integration which I created cause I had many issues with this one) then still.. why are you doing the sum? This integration does not also provide the sensor as a sum already? The second thing is you will have to subtract your battery power from inverter power when battery power is > 0 W.

**Reply by alarmatwork** [reply]
Thaks!
I will get the inverter updated to the latest firmware and will test again. 
But in my case  it does not have  a sum of all.

I am using curretly SUN-10K-SG01HP3-EU-am2.yaml would you think that using a different configuration file would expose more options?

I have only these available currently:

- AC Temperature: 32.5 °C
- Alert: ['0x0', '0x0', '0x0', '0x0', '0x0', '0x0']
- Battery Current: -0.23 A
- Battery Power: -40 W
- Battery SOC: 100%
- Battery Temperature: 28 °C
- Battery Voltage: 216.3 V
- Communication Board Version No.: 8193
- Control Board Version No.: 2844
- Current L1 : 2.9 A
- Current L2: -2.19 A
- Current L3: -1.29 A
- Daily Battery Charge: 18.3 kWh
- Daily Battery Discharge: 0.1 kWh
- Daily Energy Bought: 46.6 kWh
- Daily Energy Sold: 0 kWh
- Daily Load Consumption: 27 kWh
- Daily Production: 0 kWh
- DC Temperature: 25 °C
- External CT L1 Power: 1,882 W
- External CT L2 Power: 1,879 W
- External CT L3 Power: 1,885 W
- Grid Voltage L1: 233.9 V
- Grid Voltage L2: 235.7 V
- Grid Voltage L3: 235.4 V
- Internal CT L1 Power: 1,840 W
- Internal CT L2 Power: 1,876 W
- Internal CT L3 Power: 1,919 W
- Inverter ID: 2301048018
- Inverter L1 Power: 641 W
- Inverter L2 Power: -532 W
- Inverter L3 Power: -334 W
- Load L1 Power: 2,523 W
- Load L2 Power: 1,346 W
- Load L3 Power: 1,550 W
- Load Voltage L1: 235 V
- Load Voltage L2: 237 V
- Load Voltage L3: 236.5 V
- PV1 Current: 0.1 A
- PV1 Power: 0 W
- PV1 Voltage: 0.6 V
- PV2 Current: 0.2 A
- PV2 Power: 0 W
- PV2 Voltage: 0 V
- Status_connection: Connected
- Status_lastUpdate: 12/04/2024, 06:29:29
- Total Battery Charge: 80.4 kWh
- Total Battery Discharge: 59.3 kWh
- Total Energy Bought: 5,404.3 kWh
- Total Energy Sold: 75.2 kWh
- Total Grid Power: 5,653 W
- Total Grid Production: 75.2 kWh
- Total Load Consumption: 8,524.2 kWh
- Total Load Power: 5,427 W
- Total Power: Unavailable
- Total Production: 3,747.3 kWh

**Reply by davidrapan** [reply]
Yes, it would. Look: [davidrapan/ha-solarman/deye_sg01hp3.yaml](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/deye_sg01hp3.yaml), inverter output (sum of L1, L2 and L3) is called "Power" and "Power losses" is inverter consumption + AC/DC convertsion losses (but that's a custom sensor and specific to my integration). 😉
